Tasks/Kubernetes/src/kubernetes.ts

```diff
@@ -37,22 +38,28 @@ connection.open(tl.getInput("kubernetesServiceEndpoint")).then(
 ).catch((error) => tl.setResult(tl.TaskResult.Failed, error) );
 
 
-function run(clusterConnection: ClusterConnection, registryAuthenticationToken: RegistryAuthenticationToken)
+async function run(clusterConnection: ClusterConnection, registryAuthenticationToken: RegistryAuthenticationToken)
 {
     var secretName = tl.getInput("secretName", false);
+    var configMapName = tl.getInput("configMapName", false);
 
     if(secretName) {
-        kubectlSecret.run(clusterConnection, registryAuthenticationToken, secretName).fin(function cleanup(){
+        await kubectlSecret.run(clusterConnection, registryAuthenticationToken, secretName).fin(function cleanup(){
             clusterConnection.close();
-        }).then(function success() {
-            executeKubectlCommand(clusterConnection);
         }, function failure(err) {
             tl.setResult(tl.TaskResult.Failed, err.message);
```

Comment:
you can use try catch here. clusterConnection is not cleaned while exception occurs.